### PR TITLE
For config v3, improve the error message when failing to connect to the auth service

### LINKED ISSUE #17846: Improve error emitted by Node on join when proxy address is specified in `auth_server` with a v3 config
Expected behavior:
A nicer error that suggests that the user may have inadvertently configured this incorrectly, and that points them towards providing this value in `proxy_server` instead.

I think we should be able to detect this error, since it's pretty clear cut.

Current behavior:
```
Oct 26 13:56:58 instance teleport[30996]: 2022-10-26T13:56:58Z ERRO [PROC:1]    Node failed to establish connection to cluster: connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for *.proxy.addr, proxy.addr, not foo.teleport.cluster.local". pid:30996.1 service/connect.go:119
```

### PR DESCRIPTION
**Pull request by ryanclark**
As it stands, using config v3 and specifying an proxy URL in the `auth_server` property will result in the node failing to connect to Teleport, with something like

```
Node failed to establish connection to cluster: connection error: desc = "transport: authentication handshake failed: x509: certificate is valid for go.teleport, *.teleport, *.go.teleport, not 74656c65706f7274.teleport.cluster.local
```

in the error logs.

This adds a check after where that error is printed, checking if the config version is v3 and if so, prints a more user friendly error prompting the user to check if they've specified a proxy address in `auth_server`, and to swap it over to `proxy_server` if they have.

### CODE REVIEW

File: lib/service/connect.go [hunk 1]
```diff
@@ -117,6 +117,9 @@ func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*
 			process.log.Error("Can not join the cluster as node, the token expired or not found. Regenerate the token and try again.")
 		case connectErr != nil:
 			process.log.Errorf("%v failed to establish connection to cluster: %v.", role, connectErr)
+			if process.Config.Version == defaults.TeleportConfigVersionV3 {
```
Comment (ryanclark) [inline lib/service/connect.go:120]: Do we do a better check here? We could check the error message for `transport: authentication handshake failed:`, but I'm not sure if it's better to produce this tip more often than not

Comment (strideynet) [reply lib/service/connect.go:120]: I think we can leverage `process.Config.ProxyServer.IsEmpty()` to cut down on false positives.

**Review by strideynet — APPROVED** [review]

File: lib/service/connect.go [hunk 2]
```diff
@@ -117,6 +117,9 @@ func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*
 			process.log.Error("Can not join the cluster as node, the token expired or not found. Regenerate the token and try again.")
 		case connectErr != nil:
 			process.log.Errorf("%v failed to establish connection to cluster: %v.", role, connectErr)
+			if process.Config.Version == defaults.TeleportConfigVersionV3 && process.Config.ProxyServer.IsEmpty() {
+				process.log.Errorf("Check to see if the config has auth_server pointing to a Teleport Proxy. If it does, use proxy_server instead of auth_server.")
```
Comment (nklaassen) [inline lib/service/connect.go:121]: Is it possible to inspect the error to see if it's likely we're trying to connect to a proxy? As it stands we have no idea here, would be nice to be reasonably sure that's the problem before recommending `proxy_server`

Comment (ryanclark) [reply lib/service/connect.go:121]: I spoke with @strideynet about this, and we came to the conclusion that it's probably more hassle to check the error than not. TLS libraries on macOS emit different errors than on Linux, which would add complexity to the error check.

Comment (nklaassen) [reply lib/service/connect.go:121]: that's fine, I wasn't sure how hard/easy it would be. I will just recommend a bit softer wording

Comment (nklaassen) [reply lib/service/connect.go:121]: wait, I misread it the first time, this looks fine

**Review by nklaassen — APPROVED** [review]

**Pull request merged**